ptf/tests/common/base_test.py

```diff
@@ -381,17 +365,10 @@ def verify_packet_in(self, exp_pkt, exp_in_port, timeout=2):
 
             # Here we only compare the integer value of ingress port metadata instead
             # of the byte string.
-            if is_tna():
-                rx_inport = struct.unpack("!I", rx_in_port_)[0]
-            else:
-                pkt_in_metadata = get_controller_packet_metadata(
-                    self.p4info, meta_type="packet_in", name="ingress_port"
-                )
-                pkt_in_ig_port_bitwidth = pkt_in_metadata.bitwidth
-                rx_in_port_ = de_canonicalize_bytes(
-                    pkt_in_ig_port_bitwidth, rx_in_port_
-                )
-                rx_inport = struct.unpack("!H", rx_in_port_)[0]
+            rx_inport = 0
+            for b in rx_in_port_:
+                rx_inport <<= 8
+                rx_inport += b
 
             if exp_in_port != rx_inport:
```

Comment:
Is this equivalent? If yes, would this be easier to read?
```suggestion
            exp_in_port_ = stringify(exp_in_port)
            if canonical(exp_in_port_) != canonical(rx_inport_):
```

Reply:
We can actually get the value with `int.from_bytes` which I think it is simpler